File: In C/Pascal-Triangle.c
#include <stdio.h>

int fact(int n) {
    int f = 1;
    for(int i = 1; i <= n; i++) {
        f *= i;
    }
    return f;
}

int ncr(int n, int r) {
    return fact(n) / (fact(r) * fact(n - r));
}

int main() {
    int n;
    printf("Enter number of rows: ");
    scanf("%d", &n);

    for(int i = 0; i < n; i++) {
        for(int j = 0; j < n - i; j++) printf(" ");
        for(int j = 0; j <= i; j++) {
            printf("%d ", ncr(i, j));
        }
        printf("\n");
    }
    return 0;
}

File: In Rust/Pascal-Triangle.rs
use std::io;

fn fact(n: u32) -> u32 {
    (1..=n).product()
}

fn ncr(n: u32, r: u32) -> u32 {
    fact(n) / (fact(r) * fact(n - r))
}

fn main() {
    println!("Enter number of rows: ");
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let n: u32 = input.trim().parse().unwrap();

    for i in 0..n {
        print!("{}", " ".repeat((n - i) as usize));
        for j in 0..=i {
            print!("{} ", ncr(i, j));
        }
        println!();
    }
}
